src/EditorFeatures/CSharpTest/OnTheFlyDocs/OnTheFlyDocsUtilitiesTests.cs

```diff
@@ -110,21 +88,620 @@ struct CustomStruct
+
+    [Fact]
+    public async Task TestParametersWithMultipleLevelNestedTypes()
+    {
+        var testCode = """
+        class C
+        {
+            void ProcessNested(CustomOuter.CustomInner.DeepNested nested)
+            {
+                System.Console.WriteLine(nested.Value);
+            }
+        }
+
+        class CustomOuter
+        {
+            public class CustomInner
+            {
+                public class DeepNested
+                {
+                    public int Value { get; set; }
+                }
+            }
+        }
+        """;
+
+        var nestedText = """
+        public class DeepNested
+                {
+                    public int Value { get; set; }
+                }
+        """;
+
+        await TestMethodContextAsync(testCode, nestedText);
+    }
+
+    [Fact]
+    public async Task TestIndexerParameterTypes()
+    {
+        var testCode = """
+        class C
+        {
+            CustomCollection collection = new CustomCollection();
+            
+            void UseCollection()
+            {
+                var item = collection[new CustomKey(5)];
+            }
+        }
+
+        class CustomCollection
+        {
+            public string this[CustomKey key] => key.Id.ToString();
+        }
+
+        class CustomKey
+        {
+            public int Id { get; }
+            
+            public CustomKey(int id)
+            {
+                Id = id;
+            }
+        }
+        """;
+
+        var keyText = """
+        class CustomKey
+        {
+            public int Id { get; }
+            
+            public CustomKey(int id)
+            {
+                Id = id;
+            }
+        }
+        """;
+
+        await TestSymbolContextAsync(
+            testCode,
+            (syntaxTree, semanticModel) =>
+            {
+                var classDeclaration = syntaxTree.GetRoot()
+                    .DescendantNodes()
+                    .OfType<ClassDeclarationSyntax>()
+                    .FirstOrDefault(c => c.Identifier.Text == "CustomCollection");
+
+                var indexerDeclaration = classDeclaration!.Members
+                    .OfType<IndexerDeclarationSyntax>()
+                    .First();
+
+                return semanticModel.GetDeclaredSymbol(indexerDeclaration)!;
+            },
+            keyText);
+    }
+
+    private static async Task<(Solution Solution, Document Document, SemanticModel SemanticModel, SyntaxTree SyntaxTree)>
+        SetupWorkspaceAsync(string testCode)
+    {
+        var workspace = EditorTestWorkspace.CreateCSharp(testCode);
         var solution = workspace.CurrentSolution;
         var document = solution.Projects.First().Documents.First();
 
         var syntaxTree = await document.GetSyntaxTreeAsync();
         var semanticModel = await document.GetSemanticModelAsync();
 
-        var methodInvocation = syntaxTree!.GetRoot()
+        return (solution, document, semanticModel!, syntaxTree!);
```

Comment:
1. this needs to return the workspace.  THe caller needs to dispose it at the end.
2. no need to return document and solution, the former gives you the latter.
3. no need to return semantic model and syntax tree, the former gives you the latter.
4. tuples names are camelCased :)